### PR TITLE
dtls13: support Authentication and Integrity-Only Cipher Suites

### PR DESCRIPTION
**Pull request by rizlik**
# Description

Support Authentication and Integrity-Only Cipher Suites (TLS13-SHA256-SHA256 and TLS13-SHA384-SHA384) in DTLSv1.3 (see RFC 9150)

# Testing
A new test was added, also it can be tested using `-v4 -u -l TLS13-SHA256-SHA256` options for `./examples/server/server` and `./examples/client/client`

### CODE REVIEW

**Review by julek-wolfssl — CHANGES_REQUESTED** [review]
I like the testing you set up. Please also add the ciphers to `tests/test-dtls13.conf` like in `tests/test-tls13.conf`.

File: tests/test-dtls13.conf [hunk 1]
```diff
@@ -270,3 +270,23 @@
 -u
 -v 4
 -l TLS_AES_128_GCM_SHA256
+
+# server TLSv1.3 Integrity-only SHA256
```
Comment (julek-wolfssl) [inline tests/test-dtls13.conf:274]: `TLSv1.3` -> `DTLSv1.3`
Same in all other occurrences in this file.

File: tests/api.c [hunk 2]
```diff
@@ -65849,6 +65849,78 @@ static int test_dtls13_bad_epoch_ch(void)
...
 
+#if defined(HAVE_NULL_CIPHER) && defined(HAVE_IO_TESTS_DEPENDENCIES) && \
+    defined(WOLFSSL_DTLS13)
+static byte* test_find_string(const char *string,
+    byte *buf, int buf_size)
+{
+    int string_size, i;
+
+    string_size = XSTRLEN(string);
+    for (i = 0; i < buf_size; i++) {
```
Comment (julek-wolfssl) [inline tests/api.c:65860]: If you do `i < buf_size - string_size - 1` you can skip the size check inside the loop.

File: tests/api.c [hunk 3]
```diff
@@ -65849,6 +65849,78 @@ static int test_dtls13_bad_epoch_ch(void)
...
+        if (buf[i] != *string)
+            continue;
```
Comment (julek-wolfssl) [inline tests/api.c:65862]: Skip this. `XSTRCMP` already takes care of this.

**Review by julek-wolfssl — CHANGES_REQUESTED** [review]

File: tests/api.c [hunk 4]
```diff
@@ -65849,6 +65849,74 @@ static int test_dtls13_bad_epoch_ch(void)
...
+    XMEMSET(&test_ctx, 0, sizeof(test_ctx));
+    test_ctx.c_ciphers = test_ctx.s_ciphers = "TLS13-SHA256-SHA256";
+    ret = test_memio_setup(&test_ctx, &ctx_c, &ctx_s, &ssl_c, &ssl_s,
+        wolfDTLSv1_3_client_method, wolfDTLSv1_3_server_method);
+    if (ret != 0)
+        return TEST_FAIL;
+    ret = test_memio_do_handshake(ssl_c, ssl_s, 10, NULL);
+    if (ret != 0)
+        return TEST_FAIL;
+    test_str_size = XSTRLEN("test") + 1;
```
Comment (julek-wolfssl) [inline tests/api.c:65885]: You could just use `sizeof("test")` but this is not important for testing code.

**Review by julek-wolfssl — APPROVED** [review]

**Review by JacobBarthelmeh — APPROVED** [review]

**Pull request merged**